dapr/clients/grpc/_helpers.py

```diff
@@ -6,6 +6,98 @@
 """
 
 from typing import Dict, List, Union, Tuple
+from grpc import UnaryUnaryClientInterceptor, ClientCallDetails     # type: ignore
+
+from collections import namedtuple
 
 MetadataDict = Dict[str, List[Union[bytes, str]]]
 MetadataTuple = Tuple[Tuple[str, Union[bytes, str]], ...]
+
+
+class _ClientCallDetails(
+        namedtuple(
+            '_ClientCallDetails',
+            ['method', 'timeout', 'metadata', 'credentials', 'wait_for_ready', 'compression']),
+        ClientCallDetails):
+    """This is an implementation of the ClientCallDetails interface needed for interceptors.
+    This class takes six named values and inherits the ClientCallDetails from grpc package.
+    This class encloses the values that describe a RPC to be invoked.
+    """
+    pass
+
+
+class HeaderInterceptor(UnaryUnaryClientInterceptor):
```

Comment:
Later, this can be used by trace id injection, so that I would rename `HeaderInterceptor` to `DaprClientInterceptor`.